rock_paper_scissors/user.py

```diff
@@ -7,19 +7,18 @@ def __init__(self, name):
         self._choice = None
         self.name = name
 
+    def __repr__(self):
+        return self.name
+
     @property
     def choice(self):
         return self._choice
 
     @choice.setter
     def choice(self, value):
-        if value in VALID_CHOICES:
-            move = self.convert_input_to_move(value)
-            self._choice = move
-        else:
-            print('Invalid Input')
-            exit()
+        try:
+            assert isinstance(value, Move)
+        except AssertionError:
+            raise Exception('Invalid instance of Move')
```

Comment:
Technically this exception only gets raised if `value` is _not_ an instance of `Move`, rather than an invalid one.

It would be more correct to say "Choice must be an instance of Move" or something to that effect.